tests/src/python/test_qgsfileutils.py

```diff
@@ -58,8 +58,8 @@ def testAddExtensionFromFilter(self):
     def testStringToSafeFilename(self):
         self.assertEqual(QgsFileUtils.stringToSafeFilename('my FiLe v2.0_new.tif'), 'my FiLe v2.0_new.tif')
         self.assertEqual(
-            QgsFileUtils.stringToSafeFilename('rendered map_final? rev (12-03-1017)_real@#$&*#%&*$!!@$%^&(*(.tif'),
-            'rendered map_final_ rev _12-03-1017__real____________________.tif')
+            QgsFileUtils.stringToSafeFilename('rendered map_final? rev (12-03-1017)_real/\\?%*:|"<>.tif'),
```

Comment:
Is there a particular reason to remove `#` etc. from the test?

Reply:
Because now it's based on characters which are actually illegal - # and the others are valid characters.